src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs

```diff
@@ -960,6 +960,61 @@ void TestAction(HttpResponse httpResponse)
             Assert.Equal(httpContext.Response, httpResponseArgument);
         }
 
+        [Fact]
+        public async Task RequestDelegateSetContentTypeWhenNullAndReturnTypeIsString()
+        {
+            string Plaintext() => "Hello, World!";
+            var httpContext = new DefaultHttpContext();
+
+            var requestDelegate = RequestDelegateFactory.Create((Func<string>)Plaintext, new EmptyServiceProvider());
+
+            await requestDelegate(httpContext);
+
+            Assert.Equal(ContentTypeConstants.PlainTextContentTypeWithCharset, httpContext.Response.ContentType);
+        }
+
+        [Fact]
+        public async Task RequestDelegateSetContentTypeWhenNullAndReturnTypeIsGenericOfTypeTask()
+        {
+            async Task<string> Plaintext()
+            {
+                var sayHelloTask = Task.Run(() =>
+                {
+                    return "Hello World Again";
+                });
+
+               return await sayHelloTask;
+            }
+            var httpContext = new DefaultHttpContext();
+
+            var requestDelegate = RequestDelegateFactory.Create((Func<Task<string>>)Plaintext, new EmptyServiceProvider());
+
+            await requestDelegate(httpContext);
+
+            Assert.Equal(ContentTypeConstants.PlainTextContentTypeWithCharset, httpContext.Response.ContentType);
+        }
+
+        [Fact]
+        public async Task RequestDelegateSetContentTypeWhenNullAndReturnTypeIsGenericOfTypeValueTask()
+        {
+            async ValueTask<string> Plaintext()
+            {
+                var sayHelloTask = Task.Run(() =>
+                {
+                    return "Hello World Again";
+                });
+
+                return await sayHelloTask;
+            }
+            var httpContext = new DefaultHttpContext();
+
+            var requestDelegate = RequestDelegateFactory.Create((Func<ValueTask<string>>)Plaintext, new EmptyServiceProvider());
+
+            await requestDelegate(httpContext);
+
+            Assert.Equal(ContentTypeConstants.PlainTextContentTypeWithCharset, httpContext.Response.ContentType);
+        }
```

Comment:
We should add a test cases for callbacks that set `HttpContext.Response.ContentType` before returning a `string/Task<string>/ValueTask<string>` verifying it doesn't get overwritten.